src/transformers/visitors/orchestrationInternalFunctionCallVisitor.ts

```diff
@@ -306,22 +310,32 @@ const internalCallVisitor = {
                             childNode.body.statements[id-1] = statenode;
                            node.body.statements.forEach(kidNode =>{
                             if(kidNode.nodeType === 'ExpressionStatement'&& kidNode.expression.name === state.internalFncName[index]) {
-                               kidNode.expression = Object.assign(kidNode.expression,statenode.initialValue);
+                              if (kidNode.expression.operator) {
+                                const newExpressionNode = Object.assign(cloneDeep(kidNode.expression), statenode.initialValue);
+                                node.body.statements.push(newExpressionNode);
+                            } else {
+                                Object.assign(kidNode.expression, statenode.initialValue);
+                            }
                             }
+
                            });
-                          childNode.body.statements[id-1].initialValue =undefined;
+                          childNode.body.statements[id-1].initialValue = undefined;
                           } else{
-                            node.body.statements.forEach(kidNode =>{
-                              if(kidNode.nodeType === 'ExpressionStatement'&& kidNode.expression.name === state.internalFncName[index]) {
-                                 kidNode.expression = Object.assign(kidNode.expression,statenode.expression);
+                            node.body.statements.forEach(kidNode => {
+                              if(kidNode.nodeType === 'ExpressionStatement' && kidNode.expression.name === state.internalFncName[index]) {
+                                if(kidNode.expression.operator) {
```

Comment:
What does this bit of the code do? What does kidNode.expression.operator check? A comment maybe would be helpful here.

Reply:
added